### PR TITLE
fix(table): Fix double border on `bordered` Table Rows in `table-footer`

### PR DESCRIPTION
**Pull request by macandcheese**
**Related Issue:** #8508

## Summary
Adds a conditional class to Table Rows slotted in `table-footer` slot to prevent double border. We were doing this for body rows, but not footer rows. Adds screenshot tests to verify and updates some local samples to show this combination of props / slotted content.

| Before  | After |
| ------------- | ------------ | 
| Screenshot 2023-12-27 at 4 46 36 PM | Screenshot 2023-12-27 at 4 46 01 PM | 
Screenshot 2023-12-27 at 4 47 09 PM | Screenshot 2023-12-27 at 4 48 17 PM |
Screenshot 2023-12-27 at 4 49 07 PM | Screenshot 2023-12-27 at 4 48 37 PM |

### CODE REVIEW

File: packages/calcite-components/src/components/table/table.stories.ts [hunk 1]
```diff
@@ -1065,6 +1065,80 @@ export const tableCellCssBackgroundVariable_TestOnly = (): string =>
     </calcite-table-row>
   </calcite-table>`;
 
+export const tableBorderedWithStripedAndSingleFooter_TestOnly = (): string =>
+  html`<calcite-table striped bordered caption="Simple-bordered-with-footer table" dir="rtl" selection-mode="multiple">
```
Comment (driskull) [inline packages/calcite-components/src/components/table/table.stories.ts:1069]: should this have dir=rtl? its not in the name of the test.

File: packages/calcite-components/src/components/table/table.stories.ts [hunk 2]
```diff
@@ -1065,6 +1065,80 @@ export const tableCellCssBackgroundVariable_TestOnly = (): string =>
...
+    <calcite-table-row slot="table-footer">
+      <calcite-table-cell>foot</calcite-table-cell>
+      <calcite-table-cell>foot</calcite-table-cell>
+      <calcite-table-cell>foot</calcite-table-cell>
+      <calcite-table-cell>foot</calcite-table-cell>
+    </calcite-table-row>
+  </calcite-table>`;
+
+export const tableBorderedWithMultipleFooter_TestOnly = (): string =>
+  html`<calcite-table bordered caption="Simple-bordered-with-footer table" dir="rtl" selection-mode="multiple">
```
Comment (driskull) [inline packages/calcite-components/src/components/table/table.stories.ts:1103]: same with RTL here.

**Review by driskull — APPROVED** [review]
👍

File: packages/calcite-components/src/components/table/table.stories.ts (hunk 1 above)
Comment (macandcheese) [reply packages/calcite-components/src/components/table/table.stories.ts:1069]: Removed, thanks for catching.

File: packages/calcite-components/src/components/table/table.stories.ts (hunk 2 above)
Comment (macandcheese) [reply packages/calcite-components/src/components/table/table.stories.ts:1103]: Removed, thanks for catching.

**Pull request merged**